File: app/src/main/kotlin/com/ufkoku/ciphersharedpreferences/demo/ui/main/di/MainActivityComponent.java
package com.ufkoku.ciphersharedpreferences.demo.ui.main.di;

import com.ufkoku.ciphersharedpreferences.demo.ui.main.MainPresenter;
import com.ufkoku.ciphersharedpreferences.demo.ui.main.MainViewState;

import dagger.Component;

@Component(modules = {MainAcivityModule.class})
public interface MainActivityComponent {

    MainPresenter getMainPresenter();

    MainViewState getMainViewState();

}

File: app/src/main/kotlin/com/ufkoku/ciphersharedpreferences/demo/ui/main/di/MainActivityComponent.kt
package com.ufkoku.ciphersharedpreferences.demo.ui.main.di

import com.ufkoku.ciphersharedpreferences.demo.ui.main.MainPresenter
import com.ufkoku.ciphersharedpreferences.demo.ui.main.MainViewState

import dagger.Component

@Component(modules = [MainActivityModule::class])
interface MainActivityComponent {

    val mainPresenter: MainPresenter

    val mainViewState: MainViewState

}
